### PR TITLE
KeyVault: deprecating the PurgeDisabled flag from secret and key attributes

### PR DESCRIPTION
**Pull request by dragav**
## Description
The PurgeDisabled flag, exposed via Secret and Key attributes, is being deprecated in favor of its replacement, DeletionRecoveryLevel. This change only marks the field as obsolete, with removal expected in the subsequent release of PowerShell.

The flag is presently read-only, and its value won't change with this PowerShell release; the field has been already removed from the REST API protocol, and its value has always been 'false' with the exception of select Azure subscriptions. 
---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [X] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [X] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [X] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.
- [X] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [X] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [X] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [X] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [X] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [X] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by cormacpayne** [comment]
@dragav Hey Dragos, the build is failing because the property you are deprecating is being used in a few other files. You will need to wrap these usages: https://stackoverflow.com/questions/968293/c-sharp-selectively-suppress-custom-obsolete-warnings

**Comment by dragav** [comment]
@cormacpayne - thanks, I realized I was missing a commit, which I left out of my initial push. It should be working now. Thanks.

**Comment by dragav** [comment]
The latest build failed because of a ResMgr\common test (\ResourceManager\Common\Commands.ScenarioTests.ResourceManager.Common\EnvironmentSetupHelper.cs(409): error : Microsoft.Azure.Commands.Batch.Test.ScenarioTests.JobScheduleTests.TestJobScheduleCRUD [FAIL]).

**Comment by cormacpayne** [comment]
@azuresdkci test this please

File: src/ResourceManager/KeyVault/Commands.KeyVault/Models/KeyAttributes.cs [hunk 1]
```diff
@@ -67,12 +65,7 @@ public KeyAttributes()
 
         public DateTime? Updated { get; private set; }
 
-        /// <summary>
-        /// Obsolete flag indicating that the key is protected against purge.
-        /// </summary>
-        /// <remarks>
-        /// Deprecated, will be removed in the next PowerShell release.
-        /// </remarks>
+        [Obsolete("This property is being deprecated and will be removed in a future release")]
```
Comment (cormacpayne) [inline src/ResourceManager/KeyVault/Commands.KeyVault/Models/KeyAttributes.cs]: @dragav I would update all warnings with a message like

_"The PurgeDisabled property is being deprecated and will be removed in a future release. Please use the RecoveryLevel property instead."_

**Comment by dragav** [comment]
Thanks, Cormac, please review.

**Pull request merged**